packages/mds-ingest-service/repository/index.ts

```diff
@@ -505,6 +508,29 @@ export const IngestRepository = ReadWriteRepository.Create(
         }
       },
 
+      getH3Bins: async (params: GetH3BinOptions) => {
+        try {
+          const { start, end = now(), h3_resolution, k = K_HOURLY } = params
+
+          const connection = await repository.connect('ro')
+          const entities = await connection
+            .createQueryBuilder()
+            .select(['provider_id'])
+            .addSelect(h3_resolution, 'h3_identifier')
+            .addSelect(`COUNT(${h3_resolution})`, 'count')
+            .from('telemetry_annotations', 'annotation')
+            .where('timestamp >= :start', { start })
+            .andWhere('timestamp <= :end', { end })
+            .groupBy(`provider_id, ${h3_resolution}`)
+            .having(`COUNT(${h3_resolution}) >= :k`, { k })
+            .execute()
+
+          return entities as H3Bin[]
```

Comment:
Shouldn't need to do an `as` cast here

Reply:
with `.execute()` you do, it returns `any[]`. Doing `.getMany()` can return typed responses, but then you'd need to definite an Entity model for TypeORM to do parsing on.

Reply:
You should be able to type an `any[]` as any `T[]` without an `as` cast by just putting the desired typedef as part of the declaration of `entities`. e.g. `const { entities }: { entities: Foo[] } = someQuery.execute()`

`as` cast ain't necessary here.